nextcord/application_command.py

```diff
@@ -0,0 +1,883 @@
+
+
+class ApplicationCommandResponseOption:
+    """Represents an argument/parameter/option or subcommand of an Application Command.
+
+    Attributes
+    ----------
+    type: :class:`ApplicationCommandOptionType`
+        Enum corresponding to the Application Command Option type. (subcommand, string, integer, etc.)
+    name: :class:`str`
+        Name of the option or subcommand.
+    description: :class:`str`
+        Description of the option or subcommand.
+    required: :class:`bool`
+        If this option is required for users or not.
+    """
+
+    def __init__(self, payload: dict):
+        self.type = ApplicationCommandOptionType(int(payload["type"]))
+        self.name: str = payload['name']
+        self.description: str = payload['description']
+        self.required: Optional[bool] = payload.get('required')
+        self.choices: List[ApplicationCommandResponseOptionChoice] = self._create_choices(payload.get('choices', []))
+        self.options: List[ApplicationCommandResponseOption] = self._create_options(payload.get('options', []))
+
+    @staticmethod
+    def _create_choices(data: List[dict]) -> List[ApplicationCommandResponseOptionChoice]:
+        return [ApplicationCommandResponseOptionChoice(raw_choice) for raw_choice in data]
+
+    @staticmethod
+    def _create_options(data: List[dict]) -> List[ApplicationCommandResponseOption]:
+        return [ApplicationCommandResponseOption(raw_option) for raw_option in data]
+
+
+class ClientCog:
+    # TODO: I get it's a terrible name, I just don't want it to duplicate current Cog right now.
+    __cog_application_commands__: Dict[int, ApplicationCommand]
+    __cog_to_register__: List[ApplicationCommand]
+
+    def __new__(cls, *args, **kwargs):
+        new_cls = super(ClientCog, cls).__new__(cls)
+        new_cls._read_methods()
+        return new_cls
+
+    def _read_methods(self):
+        self.__cog_to_register__ = []
+        for base in reversed(self.__class__.__mro__):
+            for elem, value in base.__dict__.items():
+                is_static_method = isinstance(value, staticmethod)
+                if is_static_method:
+                    value = value.__func__
+                if isinstance(value, ApplicationCommand):
+                    if isinstance(value, staticmethod):
+                        raise TypeError(f"Command {self.__name__}.{elem} can not be a staticmethod.")
+                    value.cog_parent = self
+                    self.__cog_to_register__.append(value)
+                elif isinstance(value, ApplicationSubcommand):
+                    value.cog_parent = self
+
+    @property
+    def to_register(self) -> List[ApplicationCommand]:
+        return self.__cog_to_register__
+
+
+class SlashOption:
+    def __init__(self, name: str = None, description: str = None, required: bool = None, choices: dict = None,
+                 default: Any = None, channel_types: List[ChannelType, ...] = None):
```

Comment:
Is there a reason for ``...``, did you mean ``Any`` or somethin?

Reply:
I think it's a misuse of Ellipsis in tuple types: <https://mypy.readthedocs.io/en/stable/kinds_of_types.html#tuple-types>